### PR TITLE
[#4] [Android] [UI] As a user, I can see the splash screen

### LINKED ISSUE #4: [Android] [UI] As a user, I can see the splash screen
## Why

When users open the app, the mobile app should allow users to see the splash screen with the Nimble logo.

## Acceptance Criteria

- Display the status bar in white color.
- Display the image background with an overlay.
- Display the Nimble logo.
- Implement animation to show the logo based on Figma design.

## Design

Image

Link: https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A0

### PR DESCRIPTION
**Pull request by suho**
Resolves #4

## What happened 👀

- Added splash screen for the android application
- Added Jetpack Compose to the project

## Insight 📝

`N/A`

## Proof Of Work 📹

Screen Shot 2022-08-31 at 11 36 19

### CODE REVIEW

**Comment by luongvo** [comment]
@suho Should we set the base of this PR to `chore/82-improve-project-configurations` to ease code review process? 🤔

**Comment by suho** [comment]
@luongvo Sorry, I forgot to change the target branch :D

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import co.nimblehq.ic.kmm.suv.android.ui.screens.main.MainScreen
+import co.nimblehq.ic.kmm.suv.android.ui.screens.splash.SplashScreen
+
+@Composable
+fun Navigation() {
+    val navController = rememberNavController()
+    NavHost(navController = navController, startDestination = Screens.SPLASH) {
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt:13]: I think it's better to define the `navController` and start `startDestination` if you want to make it scalable, we can inject different NavController for usage.

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import co.nimblehq.ic.kmm.suv.android.ui.screens.main.MainScreen
+import co.nimblehq.ic.kmm.suv.android.ui.screens.splash.SplashScreen
+
+@Composable
+fun Navigation() {
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt:11]: OPTIONAL: `NavGraph` sounds more familiar with Android's developers 😄

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import co.nimblehq.ic.kmm.suv.android.R
+import kotlinx.coroutines.delay
+
+@Composable
+fun SplashScreen(onDidLaunch: () -> Unit) {
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:17]: `onDidLaunch` 🙄 This shows how iOS devs apply their style in the different language 🤣

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    LaunchedEffect(key1 = true) {
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:18]: Seems we don't update the value in `LaunchedEffect`, I think we could get rid of `key1  = true`

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                contentScale = ContentScale.FillWidth,
+                modifier = Modifier
+                    .matchParentSize()
+            )
+        }
+        Image(
+            painter = painterResource(id = R.drawable.ic_logo_white),
+            contentDescription = null,
+            modifier = Modifier.align(Alignment.Center)
+        )
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:36]: With the convention, we should define the `modifier` on top of params.

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    }
+}
+
+@Preview
+@Composable
+fun DefaultPreview() {
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:42]: https://nimblehq.co/compass/development/android/user-interface/jetpack-compose/

```suggestion
fun SplashScreenPreview() {
```

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.ic.kmm.suv.android.ui.theme
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Shapes
+import androidx.compose.ui.unit.dp
+
+val Shapes = Shapes(
+    small = RoundedCornerShape(4.dp),
+    medium = RoundedCornerShape(4.dp),
+    large = RoundedCornerShape(0.dp)
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt:10]: Small == Medium? 🤔

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/main/MainScreen.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import androidx.compose.ui.graphics.Color
+import co.nimblehq.ic.kmm.suv.Greeting
+
+@Composable
+fun MainScreen() {
+    Box(modifier = Modifier.fillMaxSize()) {
+        Text(
+            text = Greeting().greeting(),
+            color = Color.Black,
+            modifier = Modifier.align(Alignment.Center)
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/main/MainScreen.kt:18]: Same here. `modifier` should be on top 😄

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 4 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:18]: `key1` is required, so I can't remove this one

Screen Shot 2022-09-21 at 12 07 04

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt (hunk 7 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt:10]: @hoangnguyen92dn Yes! Just followed the Shapes class from `androidx.compose.material`

Screen Shot 2022-09-21 at 12 08 35

I will update the value later when we use it.

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 5 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:36]: @hoangnguyen92dn For `modifier`, we are using it for the `Image`/`Text` components, not define a new Compose UI, so I think we don't need to move `modifier` to the top, for example:

```kotlin
Image(
	modifier = Modifier.align(Alignment.Center),
	painter = painterResource(id = R.drawable.ic_logo_white),
	contentDescription = null
)

or
Text(
	modifier = Modifier.align(Alignment.Center),
	text = Greeting().greeting()
)
```

The examples above look not right, right? Or I'm missing sth?

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt (hunk 1 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt]: Improved by using `NavGraphBuilder`

For this project, I would prefer to use the `NavGraphBuilder` approach. I will improve this later when the android-templates define the template for compose navigation 🙌

Screen Shot 2022-09-21 at 12 31 37

Fixed in https://github.com/suho/kmm-ic/pull/79/commits/30bef640c4329f8860ab9d30b9e6134791383f58

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt (hunk 2 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/Navigation.kt]: Nice catch!

Fixed in https://github.com/suho/kmm-ic/pull/79/commits/30bef640c4329f8860ab9d30b9e6134791383f58

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 3 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt]: Lol, changed it to `onLaunch`

Fixed in https://github.com/suho/kmm-ic/pull/79/commits/30bef640c4329f8860ab9d30b9e6134791383f58

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 6 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt]: Fixed in https://github.com/suho/kmm-ic/pull/79/commits/30bef640c4329f8860ab9d30b9e6134791383f58

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/main/MainScreen.kt (hunk 8 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/main/MainScreen.kt]: Replied here: https://github.com/suho/kmm-ic/pull/79#discussion_r976041121

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 4 above)
Comment (hoangnguyen92dn) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt:18]: @suho How about replace `key1` with Unit? 😄 

```
LaunchedEffect(Unit)
```

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 5 above)
Comment (hoangnguyen92dn) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt]: @suho 🤝

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt (hunk 7 above)
Comment (hoangnguyen92dn) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt:10]: It's confused a bit for new comers IMO 😄 I'd suggest to remove it if it's not being in use. It's up to you ✋

Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/theme/Shape.kt]: Fixed in https://github.com/suho/kmm-ic/pull/79/commits/f734fc016f53c04d98972caa71f5415472473153

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt (hunk 4 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/splash/SplashScreen.kt]: Fixed in https://github.com/suho/kmm-ic/pull/79/commits/f734fc016f53c04d98972caa71f5415472473153

**Review by hoangnguyen92dn — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]
LGTM!

**Pull request merged**